arc_cache.go

```diff
@@ -105,13 +88,38 @@ func (b *arccache) GetSize(k cid.Cid) (int, error) {
 	return blockSize, err
 }
 
+func (b *arccache) View(k cid.Cid, callback func([]byte) error) error {
+	// shortcircuit and fall back to Get if the underlying store
+	// doesn't support Viewer.
+	if b.viewer == nil {
+		blk, err := b.Get(k)
+		if err != nil {
+			return err
+		}
+		return callback(blk.RawData())
+	}
+
+	if !k.Defined() {
+		log.Error("undefined cid in arc cache")
+		return ErrNotFound
+	}
+
+	if has, _, ok := b.queryCache(k); ok && !has {
```

Comment:
if the cache does have it, should we return the cached item rather than calling back to view?

Reply:
The cache does not value, just metadata about the block (whether it exists in the underlying store, and the size if it has been queried before). So unfortunately, we can't return the value here.